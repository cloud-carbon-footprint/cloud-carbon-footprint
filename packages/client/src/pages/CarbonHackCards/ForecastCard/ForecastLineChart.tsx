import React from 'react'
import ReactApexChart from 'react-apexcharts'

export class ForecastLineChart extends React.Component<
  // eslint-disable-next-line @typescript-eslint/ban-types
  { data: object; categories: object },
  { series: object[]; options: object }
> {
  constructor(props: any) {
    super(props)
    const cropped_data = []
    const cropped_category = []
    let index = 0
    props.data.map((data) => {
      if (index % 4 == 0)
        cropped_data.push((Math.round(data * 100) / 100).toFixed(2))
      index++
    })
    index = 0
    props.categories.map((category) => {
      if (index % 4 == 0) cropped_category.push(new Date(category))
      index++
    })
    this.state = {
      series: [
        {
          data: cropped_data,
        },
      ],
      options: {
        chart: {
          height: 1000,
          type: 'line',
          zoom: {
            enabled: false,
          },
        },
        dataLabels: {
          enabled: false,
        },
        stroke: {
          curve: 'straight',
          width: 0.5,
        },
        title: {
          text: '',
          // align: 'left',
        },
        grid: {
          row: {
            colors: ['#f3f3f3', 'transparent'], // takes an array which will be repeated on columns
            opacity: 0.5,
          },
        },
        xaxis: {
          title: {
            text: 'Optimal time',
            offsetX: 0,
            offsetY: 200,
          },
          categories: cropped_category.map(
            (category) =>
              category.toString().split(' ')[1] +
              ' ' +
              category.toString().split(' ')[2] +
              '; ' +
              category.toString().split(' ')[4].slice(0, -3),
          ),
        },
        yaxis: {
          title: {
            text: 'Emission rating (g/kWh)',
          },
        },
        markers: {
          size: 4,
          fillOpacity: 1,
          shape: 'circle',
        },
      },
    }
  }

  render() {
    return (
      <div id="chart">
        <ReactApexChart
          options={this.state.options}
          series={this.state.series as any}
          type="area"
          height={350}
        />
      </div>
    )
  }
}
