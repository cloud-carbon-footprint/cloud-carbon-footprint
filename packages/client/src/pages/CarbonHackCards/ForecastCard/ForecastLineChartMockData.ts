export const data = [
  449.58751306261104, 450.54823382833325, 451.59685472680553, 452.6168607867777,
  453.64569146638877, 454.5744221208333, 455.5191547521666,
]

export const categories = [
  '2022-11-08 15:40:00',
  '2022-11-08 15:45:00',
  '2022-11-08 15:50:00',
  '2022-11-08 15:55:00',
  '2022-11-08 16:00:00',
  '2022-11-08 16:05:00',
  '2022-11-08 16:10:00',
]
