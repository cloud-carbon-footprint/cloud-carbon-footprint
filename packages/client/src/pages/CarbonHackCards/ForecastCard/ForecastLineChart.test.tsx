/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react'
import { create } from 'react-test-renderer'
import { ForecastLineChart } from './ForecastLineChart'
import { data, categories } from './ForecastLineChartMockData'

jest.mock('react-apexcharts')

describe('ForecastLineChart', () => {
  it('renders line chart', async () => {
    const root = create(
      <ForecastLineChart data={data} categories={categories} />,
    )
    expect(root.toJSON()).toMatchSnapshot()
  })
})
