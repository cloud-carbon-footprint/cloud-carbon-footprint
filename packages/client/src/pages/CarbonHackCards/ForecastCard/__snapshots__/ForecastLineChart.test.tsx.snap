// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastLineChart renders line chart 1`] = `
<div
  id="chart"
/>
`;
